unstructured/ingest/connector/pinecone.py

```diff
@@ -0,0 +1,126 @@
+import json
+import multiprocessing as mp
+import typing as t
+from dataclasses import dataclass
+
+from unstructured.ingest.error import DestinationConnectionError, WriteError
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    BaseSessionHandle,
+    ConfigSessionHandleMixin,
+    WriteConfig,
+    WriteConfigSessionHandleMixin,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+@requires_dependencies(["pinecone"], extras="pinecone")
+def create_pinecone_object(api_key, index_name, environment):
```

Comment:
Would make sense for this to be a method of the `PineconeWriteConfig` class. Just an idea.

Reply:
Agreed, addressed with [test and session creation updates](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/84e65e5389f05c57508311755c837d8d0b0815dc)